src/AccessibilityInsights.Extensions.GitHub/ConfigurationModelControl.xaml.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using AccessibilityInsights.CommonUxComponents.Dialogs;
+using AccessibilityInsights.Extensions.Interfaces.IssueReporting;
+using Newtonsoft.Json;
+
+namespace AccessibilityInsights.Extensions.GitHub
+{
+    /// <summary>
+    /// Interaction logic for ConfigurationModel.xaml
+    /// </summary>
+    public partial class ConfigurationModel : IssueConfigurationControl
+    {
+        public ConnectionConfiguration Config { get; set; }
+
+        public Action<bool> IsConfigured { get; set; }
+
+        public ConfigurationModel()
+        {
+            InitializeComponent();
+        }
+
+        /// <summary>
+        /// Called when save button clicked.
+        /// </summary>
+        /// <returns>The extension’s new configuration, serialized</returns>
+        public override string OnSave()
+        {
+            if (LinkValidator.IsValidGitHubRepoLink(this.tbURL.Text))
+            {
+                this.Config.RepoLink = this.tbURL.Text;
+                canSave = false;
+                UpdateSaveButton();
+                IsConfigured(true);
+            }
+            else
+            {
+                MessageDialog.Show(Properties.Resources.InvalidLink);
+            }
+            return JsonConvert.SerializeObject(this.Config);
+        }
+
+        /// <summary>
+        /// Called when settings page dismissed
+        /// </summary>
+        public override void OnDismiss()
+        {
+            this.tbURL.Text = this.Config.RepoLink;
+            canSave = false;
+            UpdateSaveButton();
+        }
+
+        public void TextChangeUpdateSaveButton(object sender, EventArgs e)
+        {
+            TextChangeUpdateSaveButtonHelper();
+        }
+
+        public void TextChangeUpdateSaveButtonHelper()
+        {
+            string curURL = this.tbURL.Text;
+            if (!string.IsNullOrEmpty(curURL) && ((this.Config != null && string.IsNullOrEmpty(this.Config.RepoLink)) || !this.Config.RepoLink.Equals(curURL, StringComparison.InvariantCulture)))
+            {
+                canSave = true;
+            }
+            else
+            {
+                canSave = false;
+            }
+            UpdateSaveButton();
+        }
+
+        private bool canSave = false;
+        /// <summary>
+        /// Can the save button be clicked
+        /// </summary>
+        public override bool CanSave => canSave;
```

Comment:
I remember discussing moving `UpdateSaveButton` into the setter of the `CanSave` property so that we don't need to keep `canSave` changes and `UpdateSaveButton()` calls together explicitly.  Did that approach run into issues?

Reply:
I replied to the comment in the PR, the thing was CanSave has no setter by design, so we can't apply that approach

Reply:
Got it, I see - I am not sure how or if we can make it private, will take a look afterwards as a separate issue